src/clients/SpokePoolClient.ts

```diff
@@ -591,12 +596,18 @@ export class SpokePoolClient extends BaseAbstractClient {
           .flat()
```

Comment:
We only need the above `blockNumbers` array in the UBA world, so it might be nice to relocate into the `isUBAActivated` if statement. Then we'll avoid parsing over the set of events and filtering their block numbers for uniqueness as well.

Reply:
good point done

Reply:
Sorry I forgot to push a commit I had ready can you re review latest? its a simplifying refactor